### PR TITLE
feat(issue-details): Add analytics events for issue tracking

### LINKED ISSUE #46422: Track external issues being created
We have no analytics for a user creating an external issue (ex: Jira). We should add the following analytics events to the front-end:

* integrations.external_issue_modal_opened
* integrations.external_issue_created

We need to make sure to record the `provider`, `group_id`, and `project_id`.

### PR DESCRIPTION
**Pull request by malwilley**
Closes https://github.com/getsentry/sentry/issues/46422

### CODE REVIEW

File: static/app/components/group/externalIssueForm.tsx [hunk 1]
```diff
@@ -69,6 +72,14 @@ export default class ExternalIssueForm extends AbstractExternalIssueForm<Props,
   onSubmitSuccess = (_data: IntegrationExternalIssue): void => {
     const {onChange, closeModal} = this.props;
     const {action} = this.state;
+
+    trackAdvancedAnalyticsEvent('issue_details.external_issue_created', {
+      organization: this.props.organization,
+      ...getAnalyticsDataForGroup(this.props.group),
+      external_issue_provider: this.props.integration.provider.key,
+      external_issue_type: 'integration',
```
Comment (scefali) [inline static/app/components/group/externalIssueForm.tsx:80]: @malwilley can we call it `first_party`? That's the standard we follow in other places. See `IntegrationType`

File: static/app/utils/analytics/issueAnalyticsEvents.tsx [hunk 2]
```diff
@@ -15,6 +15,11 @@ type SourceMapDebugParam = {
 
 interface GroupEventParams extends CommonGroupAnalyticsData, BaseEventAnalyticsParams {}
 
+interface ExternalIssueParams extends CommonGroupAnalyticsData {
+  external_issue_provider: string;
+  external_issue_type: 'plugin' | 'integration';
```
Comment (scefali) [inline static/app/utils/analytics/issueAnalyticsEvents.tsx:20]: @malwilley we should use `IntegrationType` here

File: static/app/components/group/externalIssueForm.tsx (hunk 1 above)
Comment (malwilley) [reply static/app/components/group/externalIssueForm.tsx]: Thanks for pointing this out - didn't know about that! Changed to `IntegrationType`

Out of curiosity what's the difference between `first_party` and `sentry_app`?

Comment (scefali) [reply static/app/components/group/externalIssueForm.tsx]: @malwilley first_party means we built it, sentry_app means it's on the integration platform and a 3rd party built it

File: static/app/components/group/externalIssueActions.tsx [hunk 3]
```diff
@@ -62,11 +66,21 @@ const ExternalIssueActions = ({configurations, group, onChange}: Props) => {
     });
   };
 
-  const doOpenModal = (integration: GroupIntegration) =>
+  const doOpenModal = (integration: GroupIntegration) => {
+    trackAdvancedAnalyticsEvent('issue_details.external_issue_modal_opened', {
+      organization,
+      ...getAnalyticsDataForGroup(group),
+      external_issue_provider: integration.provider.key,
+      external_issue_type: 'first_party',
```
Comment (scefali) [inline static/app/components/group/externalIssueActions.tsx:74]: @malwilley what about the `sentry_app` case?

Comment (malwilley) [reply static/app/components/group/externalIssueActions.tsx:74]: Good catch, there was one extra component I forgot to add to

**Review by scefali — APPROVED** [review]

**Pull request merged**